src/java_bytecode/java_object_factory.h

```diff
@@ -10,18 +10,24 @@ Author: Daniel Kroening, kroening@kroening.com
 #ifndef CPROVER_JAVA_BYTECODE_JAVA_OBJECT_FACTORY_H
 #define CPROVER_JAVA_BYTECODE_JAVA_OBJECT_FACTORY_H
 
+
+#include <memory>
+
 #include <util/message.h>
 #include <util/std_code.h>
 #include <util/symbol_table.h>
 
+#include <java_bytecode/select_pointer_type.h>
+
 exprt object_factory(
   const typet &type,
   const irep_idt base_name,
   code_blockt &init_code,
   bool allow_null,
   symbol_tablet &symbol_table,
   size_t max_nondet_array_length,
-  const source_locationt &);
+  const source_locationt &,
+  std::shared_ptr<select_pointer_typet> pointer_type_selector);
```

Comment:
I have done this - I'm no longer sure about overloading the method - I feel like it has all the same draw backs of a default value (my main issue with it is can often get confusing resolution in C++ due to all of the 5 final parameters, though have distinct types, are all interchangeable).  I've done it as a standalone commit, but my feeling is I should drop it.